lib/spack/spack/fetch_strategy.py

```diff
@@ -752,11 +752,14 @@ def source_id(self):
         return self.commit or self.tag
 
     def mirror_id(self):
```

Comment:
Could toy add some comments here saying why this was modified? When we will merge upstream, I will help.

Reply:
Why?

See the linked PR above… we have `libsonata` and `py-libsonata` where one requires submodules checked out and the other does not. Yet it seems like for Spack, the sources get mirrored under the same path. So, if `py-libsonata` is used to create the source mirror, all is well, since it requires submodules. But if `libsonata` is used to create the source mirror, the submodules are not checked out and `py-libsonata` will fail to build.

I've just added some additional stuff to the mirror identifier trying to prevent this.

Reply:
Hope this is clearer? If not, we can Zoom tomorrow…

Reply:
Ok clear now 👍